### PR TITLE
refactor!!!!

### PR DESCRIPTION
**Pull request by sckott**
@vortexing 

SO there's a ton of changes here. I don't intend for you to look over all the details - but rather the high level changes, and here they are:

- added gh actions workflows to build pkg docs (lives at https://getwilds.org/rcromwell/) and run r cmd check
- pkg has tests now, minimal though, using my pkg vcr so that we're using test fixtures so tests can run quickly and not require a cromwell server to be alive after the first run
- a lot of style changes including function names, function parameter names, and variable names within functions - goal being that we follow https://design.tidyverse.org/ tidy design principles more or less
- added a getting started vignette
- configuring the url has changed, made a new function that sets the url, and whether the pkg prints messages or not about what its doing - in addition the pkg internaally keeps track of time zone so we only have to set it in one place
- other changes more minor

### CODE REVIEW

File: README.Rmd [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+remotes::install_github("getwilds/rcromwell")
+```
+
+Install a specific release version (in this case v1.0) by:
+
+```r
+remotes::install_github('getwilds/rcromwell@v1.0')
+```
+
+## Set up your Cromwell Server
```
Comment (vortexing) [inline README.Rmd:38]: We should remove this b/c it's FH specific.

File: README.md [hunk 2]
```diff
@@ -1,92 +1,145 @@
...
 
-```{r}
-## Set your Cromwell URL
-setCromwellURL(nodeAndPort = "gizmoXXX:20202")
+Set your Cromwell URL
+
+```r
+library(rcromwell)
+library(dplyr)
+cromwell_config("http://gizmoXXX:20202")
```
Comment (vortexing) [inline README.md:42]: Change to something vanilla: `cromwell_config("http://localhost:20202")` or some such nonsense.

File: docker/dockerfile [hunk 3]
```diff
@@ -8,6 +8,6 @@ RUN apt-get install -y libssh-dev
 # Install required packages
 RUN R -e "install.packages('remotes', dependencies = TRUE)"
 # Install from github
-RUN R -e "remotes::install_github('FredHutch/fh.wdlR@v2.0.3', dependencies = TRUE)"
+RUN R -e "remotes::install_github('getwilds/rcromwell@v2.0.3', dependencies = TRUE)"
```
Comment (vortexing) [inline docker/dockerfile:11]: this will need to shift to the current release version when you have one - 'getwilds/rcromwell@v1.0' or whatever

**Review by vortexing — APPROVED** [review]
Love it.  Full support.  I'll think more on tests and such in the future but that shouldn't impact moving this forward!!  Amazing work and thank you for doing this overhaul!!!!

**Comment by sckott** [comment]
Thanks for having a look @vortexing !

File: README.Rmd (hunk 1 above)
Comment (sckott) [reply README.Rmd:38]: remove the set Cromwell URL step? but we need to set a URL for the package to work. Maybe you mean don't include any PROOF related stuff?

File: README.md (hunk 2 above)
Comment (sckott) [reply README.md:42]: will do

File: docker/dockerfile (hunk 3 above)
Comment (sckott) [reply docker/dockerfile:11]: will do

**Comment by sckott** [comment]
@vortexing note that a number of issues https://github.com/getwilds/rcromwell/issues will be closed when this merges to main

File: README.Rmd (hunk 1 above)
Comment (vortexing) [reply README.Rmd:38]: The link to the DIY-cromwell-server is FH specific.  So remove this H2 and the stuff below it that is FH specific is all I meant. ;)

**Pull request merged**